### PR TITLE
[10.0] Add field 'UNECE Due date' on taxes and corresponding nomenclature

### PR DESCRIPTION
**Pull request by alexis-via**
This is required for Factur-X support, cf this PR https://github.com/OCA/edi/pull/22

Improve search view on unece code list.

### CODE REVIEW

File: account_tax_unece/models/account_tax.py [hunk 1]
```diff
@@ -27,3 +27,15 @@ class AccountTax(models.Model):
...
+        'unece.code.list', string='UNECE Due Date',
+        domain=[('type', '=', 'date')], ondelete='restrict',
+        help="Select the due date of that tax from the official "
+        "nomenclature of the United Nations Economic "
+        "Commission for Europe (UNECE), DataElement 2005. For a "
+        "sale VAT tax, it is the date on which that VAT is due to the "
+        "fiscal administration. For a purchase VAT tax, it is the date "
+        "on which that VAT can be deducted.")
+    unece_due_date_code = fields.Char(
+        related='unece_due_date_id.code', store=True,
```
Comment (pedrobaeza) [inline account_tax_unece/models/account_tax.py]: Make it `readonly=True` or you will have undesired writes.

File: account_tax_unece/models/account_tax.py [hunk 2]
```diff
@@ -27,3 +27,15 @@ class AccountTax(models.Model):
     unece_categ_code = fields.Char(
         related='unece_categ_id.code', store=True,
         string='UNECE Category Code')
+    unece_due_date_id = fields.Many2one(
+        'unece.code.list', string='UNECE Due Date',
+        domain=[('type', '=', 'date')], ondelete='restrict',
+        help="Select the due date of that tax from the official "
+        "nomenclature of the United Nations Economic "
```
Comment (pedrobaeza) [inline account_tax_unece/models/account_tax.py:34]: Align this at the level of the first `"`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Change version number

**Comment by alexis-via** [comment]
I got the info that probably other changes in nomenclature part of the Factur-X standard will take place. So this PR should be merged when the final version of the standard is published (scheduled for September 30th)

**Comment by alexis-via** [comment]
This is now ready to be merged. I updated all the UNECE nomenclatures and the Factur-X standard has been frozen.

**Comment by alexis-via** [comment]
Conflict resolved. Ready to merge.

**Review by bealdav — APPROVED** [review]

File: account_tax_unece/models/account_tax.py (hunk 2 above)
Comment (alexis-via) [reply account_tax_unece/models/account_tax.py:34]: flake8 is happy with it. I think it's a matter of taste here...

**Review by rvalyi — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**